src/client/pages/Offer/Introduction/Sidebar/index.tsx

```diff
@@ -0,0 +1,279 @@
+
+import { useTextKeys } from 'utils/textKeys'
+import { useCurrentLocale } from 'l10n/useCurrentLocale'
+
+import {
+  LARGE_SCREEN_MEDIA_QUERY,
+  SMALL_SCREEN_MEDIA_QUERY,
+} from 'utils/mediaQueries'
+import { isBundle, isNorwegian } from 'pages/OfferNew/utils'
+import { TOP_BAR_Z_INDEX } from 'components/TopBar'
+
+import { StartDate } from '../../../OfferNew/Introduction/Sidebar/StartDate'
+import { StickyBottomSidebar } from '../../../OfferNew/Introduction/Sidebar/StickyBottomSidebar'
+import { CampaignCodeModal } from './CampaignCodeModal'
+
+const SIDEBAR_WIDTH = '26rem'
+const SIDEBAR_SPACING_LEFT = '2rem'
+
+const Wrapper = styled.div`
+  width: calc(${SIDEBAR_WIDTH} + ${SIDEBAR_SPACING_LEFT});
+  max-width: 100%;
+  z-index: ${TOP_BAR_Z_INDEX};
+
+  ${LARGE_SCREEN_MEDIA_QUERY} {
+    padding-left: ${SIDEBAR_SPACING_LEFT};
+    flex-shrink: 0;
+  }
+`
+
+const Container = styled.div`
+  width: ${SIDEBAR_WIDTH};
+  max-width: 100%;
+  padding: 1.25rem 1rem 1rem 1rem;
+  margin-top: 3rem;
+  background-color: ${colorsV3.white};
+  border-radius: 8px;
+  box-shadow: 0 8px 16px rgba(0, 0, 0, 0.08);
+
+  ${LARGE_SCREEN_MEDIA_QUERY} {
+    position: fixed;
+    top: 8rem;
+    margin-top: 0;
+  }
+`
+
+const DiscountInfo = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: flex-start;
+  flex-wrap: wrap;
+  align-items: flex-start;
+  margin-bottom: 0.5rem;
+`
+
+const DiscountBadge = styled(Badge)`
+  &:not(:last-child) {
+    margin-right: 0.5rem;
+    margin-bottom: 0.5rem;
+  }
+`
+
+const Header = styled.div`
+  position: relative;
+  width: 100%;
+  display: flex;
+  align-items: baseline;
+  justify-content: space-between;
+  margin-bottom: 1.5rem;
+  flex-wrap: wrap;
+
+  ${SMALL_SCREEN_MEDIA_QUERY} {
+    flex-wrap: nowrap;
+  }
+`
+
+const Title = styled.h3`
+  width: 100%;
+  margin: 0 1rem 0 0;
+  font-size: 1.375rem;
+  line-height: 1.625rem;
+  white-space: pre-line;
+  color: ${colorsV3.gray900};
+
+  ${LARGE_SCREEN_MEDIA_QUERY} {
+    font-size: 1.5rem;
+    line-height: 2rem;
+  }
+`
+
+const Body = styled.div`
+  margin-bottom: 2.5rem;
+  font-size: 0.875rem;
+  line-height: 1.25rem;
+`
+
+const BodyTitle = styled.div`
+  margin-bottom: 0.75rem;
+  color: ${colorsV3.gray900};
+`
+
+const Footer = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  width: 100%;
+`
+
+const FooterExtraActions = styled.div`
+  width: 100%;
+  display: flex;
+  flex-direction: row;
+  justify-content: center;
+  align-items: center;
+  margin-top: 0.75rem;
+  margin-bottom: 0.25rem;
+`
+export type SidebarProps = {
+  quoteCartId: string
+  offerData: OfferData
+  refetchOfferData: () => Promise<void>
+  onCheckoutOpen: () => void
+}
+
+export const Sidebar: React.FC<SidebarProps> = ({
+  quoteCartId,
+  offerData,
+  refetchOfferData,
+  onCheckoutOpen,
+}) => {
+  const textKeys = useTextKeys()
+  const { isoLocale: locale } = useCurrentLocale()
```

Comment:
nit: why rename? I think it's clearer calling it "isoLocale"

Reply:
Just so I can call the mutation like this:
```
  const {
    data: campaignData,
    refetch: refetchAppliedCampaign,
  } = useAppliedCampaignQuery({
    variables: { quoteCartId, locale },
  })
```
Instead of
```
  const {
    data: campaignData,
    refetch: refetchAppliedCampaign,
  } = useAppliedCampaignQuery({
    variables: { quoteCartId, locale: isoLocale },
  })
```
But I agree with you. Gonna revert it.